src/components/SchemaTree.tsx

```diff
@@ -32,9 +38,30 @@ export const SchemaTree = observer<ISchemaTree>(props => {
     onGoToRef,
   };
 
+  const [selectedProps, setSelectedProps] = useState((props.maskProps || []) as Array<{
+    path: string;
+    required: any;
+  }>);
+
   const rowRenderer = React.useCallback(
-    (node, rowOptions) => <SchemaRow node={node} rowOptions={rowOptions} {...itemData} />,
-    [itemData.count],
+    (node, rowOptions) => {
+      return (
+        <SchemaRow
+          maskControls={() => (
```

Comment:
@P0lip can you add some suggestions on how do you think this should be done. Will be easier to process.

Reply:
Custom `renderRow` function. There is a similar case in tree-list and other components.

Reply:
What I mean is that we should make JSV more flexible & provide the support for allOf.
By support for allOf, I mean exposing mapped paths, as right now there is no way to determine the path of the original property.

Masking components could be placed in Studio.

Reply:
@P0lip, so do you mean, that `MaskControls` component should be placed in Studio (together with its state - `[selectedProps, setSelectedProps]` and passed to JSV through JSE ? Or do you mean that the whole `rowRenderer` should be composed in Studio? I'm not sure if the second is an option since it would mean moving `SchemaRow` component to Studio too?

Reply:
>  Or do you mean that the whole rowRenderer should be composed in Studio?

Yeah, rowRenderer prop should be exposed in JSV and the rest should be composed in Studio.
You can export SchemaRow from JSV, similarly to tree-list and its default row.